### PR TITLE
[GUVNOR-2962][GUVNOR-2990] fix missing range check

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/checks/SingleRangeCheck.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright 2005 JBoss Inc
```
Comment (Rikkola) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/checks/SingleRangeCheck.java]: Not sure why, but the year is 2005.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/cache/RuleInspectorCache.java [hunk 2]
```diff
@@ -192,10 +223,4 @@ public RuleInspector getRuleInspector( final int row ) {
     public AnalyzerConfiguration getConfiguration() {
         return configuration;
     }
-
-    public interface Filter {
-
-        boolean accept( final RuleInspector ruleInspector );
-
-    }
```
Comment (Rikkola) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/cache/RuleInspectorCache.java]: Removing this is a good move.

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/checks/base/CheckFactory.java [hunk 3]
```diff
@@ -67,8 +66,6 @@ public CheckFactory( final AnalyzerConfiguration configuration ) {
                                                                        configuration  ),
                                       new DetectDeficientRowsCheck( ruleInspector,
                                                      configuration ),
-                                      new RangeCheck( ruleInspector,
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-core/src/main/java/org/drools/workbench/services/verifier/core/checks/base/CheckFactory.java]: Hi @mariofusco just question, why the new `SingleRangeCheck` is not registered on this place but in the `RuleInspectorCache`?

**Comment by mariofusco** [comment]
@jomarko Because all the checks here work with a different logic: they run once per rule/row and check that single row. Conversely my new check runs just once for the whole dtable taking in input all the rows and checking if there are missing ranges. In this is way it is not only way faster but even more consistent and easier to implement. In general indeed it doesn't make sense to ask if a single row is causing a missing range: the missing range is the result of the fact that all the rows in the dtable (taken as a whole) aren't covering the entire domain and then it needs to be checked at dtable level.

**Comment by jomarko** [comment]
@mariofusco Thanks for the answer. It makes sense definitely.

**Pull request merged**